### DISCUSSION TITLE
Next JS: Add config option for global imports on rendered file

### DISCUSSION THREAD

**Discussion by SamuelQuinones** [markdoc/markdoc#127] [Ideas]
This may seem a bit convoluted but hear me out;

I am working on a site where the primary pages will be tsx / ts pages and I plan to use markdoc for the blog portion.

With a config option in the `markdoc/nextjs` *withMarkdoc* function to specify global imports, it could lead to smaller overall bundle sizes for use cases where only a portion of the site uses markdoc. Which I understand is probably a pretty niche use case but I think still has merit.

example of what that could look like:

```js
module.exports = withMarkdoc({
  mode: "static",
  // these imports are relative to the next.config.js location in this example and includes one local file and one node_modules file
  globalImports: [
    "../src/styles/code-highlight.css",
    "prismjs/components/jsx.min.js",
  ],
})({
  pageExtensions: // [...](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions)
});
```

And once the array of imports resolves to the correct location relative to the output file, they could just be added at the top:

```diff
// ... previous code
  const result = `import React from 'react';
import yaml from 'js-yaml';
+ ${resolvedGlobalImports.map(importString => `'import ${importString}'`)}
// renderers is imported separately so Markdoc isn't sent to the client
import Markdoc, {renderers} from '@markdoc/markdoc'
import {getSchema} from '${normalize(await resolve(__dirname, './runtime'))}';
// ... rest of code
```
this is a crude example but I think it gets the point across

This way you could offload features that only relate to your markdown pages onto the markdown pages, without including them in the rest of the app.

The primary motivation for this was in my own app - when including prismjs globally, all of my pages first load size increased by 10kb - which is not a whole lot, but realistically not all of those pages need the prismjs js code or the css for code highlighting, so being able to offload them would be ideal.

This is a pretty rough idea but I think it could be useful if fleshed out.

**Comment by mfix-stripe** [comment]
@SamuelQuinones I think you could handle this with a regular Webpack plugin for Next.js: https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config. We want to keep the `@markdoc/next.js` plugin Markdoc-focused, and handle other build features such as this one elsewhere.
